### DISCUSSION TITLE
Change build minification to keep function names for better flamegraph/callstacks

### DISCUSSION THREAD

**Discussion by GiantRobots** [twilio-labs/paste#1891] [Ideas] [answered]
When profiling my application Paste component calls (which become a pretty large number of components) component names become anonymous (see pic below)

<img width="957" alt="Screen Shot 2021-09-27 at 3 44 11 PM" src="https://user-images.githubusercontent.com/1903839/134990454-695b2940-e07f-4851-85b8-33fec02b838b.png">

I'm pretty sure if we update the config below we can keep our function names.

https://github.com/twilio-labs/paste/blob/1bcb8b3093920fdd871f8a13b498a8eb99201200/packages/paste-core/core-bundle/tools/minify.js#L12

```typescript
const result = await Terser.minify(fs.readFileSync(filePath, 'utf8'), { keep_fnames:true });
```

**Comment by TheSisb** [comment, marked as answer]
Hi @GiantRobots,

Thanks for the suggestion! This is tricky; we want to ship the absolute smallest bundle for developers using Paste, but at the same time understand the debugging concern.

My ideal approach would be to ship two versions: minified (`*.min.js`) and an unminified (`*.js`) version of our packages. While I can easily build both versions in our dist folders, I'm not sure yet how developers can toggle between both versions. What do you think?

**Reply by GiantRobots** [reply]
I thought (and we should double check the assumption) That when built in production mode webpack (and other bundlers) minifies all the assets when making chunks so when a project is ran in production mode the code ends up minified either way.

**Reply by GiantRobots** [reply]
https://stackoverflow.com/a/37901120/6190714

Seems to indicate that minification runs on dependencies so we should be gucci to keep fnames may want to keep classnames as well, or just not minify.

Here's a better example of where this becomes a problem. If I hover over the name of the file I get a path but not much more information than that making it difficult to profile my application and speed things up in the appropriate places.

<img width="1919" alt="Screen Shot 2021-09-28 at 10 52 55 AM" src="https://user-images.githubusercontent.com/1903839/135133056-168c2d3a-7ff6-4e1a-90c7-c319d09cd13a.png">

**Reply by TheSisb** [reply]
After some discussion and tests internally, we're going to distribute a min/non-minified dist JS in our packages.  A ticket has been made into our backlog. We will report back when the work has been completed. Thanks for the feedback!

**Reply by gloriliale** [reply]
The work on this has been completed and merged! Let us know if it helps!
